src/util.cc

```diff
@@ -622,3 +624,50 @@ bool Truncate(const string& path, size_t size, string* err) {
   }
   return true;
 }
+
+/// ReplaceContent the content of file_dst with new_content.
+/// The file pointed to by file_dst will keep the same uid and gid.
+/// In case of an error, false is returned and *err will carry the error message.
+bool ReplaceContent(const string& file_dst, const string& new_content, string* err) {
+#ifndef _WIN32
+  struct stat old_file;
+  bool found_uid_gid = true;
+
+  if (stat(file_dst.c_str(), &old_file) < 0) {
+    found_uid_gid = false;
+  }
+
+  if (unlink(file_dst.c_str()) < 0) {
+    *err = strerror(errno);
+    return false;
+  }
+
+  if (rename(new_content.c_str(), file_dst.c_str()) < 0) {
+    *err = strerror(errno);
+    return false;
+  }
+
+  // apply uid and gid again so we always stay as the uid and gid of the first caller that created a file
+  if (found_uid_gid) {
+    if (chown(file_dst.c_str(), old_file.st_uid, old_file.st_gid)) {
+      *err = strerror(errno);
+      return false;
+    }
+  }
+#else
+   /// This will not carry over uid / gid stats
+   /// However, this is only in windows, on linux uid / gid will be preserved
```

Comment:
Please use `//` instead of `///`.

Also explain briefly why we don't do this on Windows (you can link to your comment in the previous PR).